### ISSUE TITLE
"((" is a left paren space violation?

### ISSUE THREAD

**Issue by peterhurford** [jimhester/lintr#259]
Currently this is a violation of the `spaces_left_parentheses_linter`:

```R
if ((var == "") | (other_var == "")) {
```

I think `((` should be legal. Is there a reason it isn't?

If this is a mistake, the solution would be to amend https://github.com/jimhester/lintr/blob/master/R/spaces_left_parentheses_linter.R#L27 to include `(`, right?

**Comment by wlandau** [comment]
Such a hilarious runaround:

```r
lint("generate.R")
```

```r
## generate.R:289:10: style: Place a space before left parenthesis, except in a function call.
##   while ((out <- stri_rand_strings(1, n)) %in% exclude){
##          ^
```

After adding a space: 

```r
lint("generate.R")
```

```r
## generate.R:289:10: style: Do not place spaces around code in parentheses or square brackets.
##   while ( (out <- stri_rand_strings(1, n)) %in% exclude){
##         ^
```

Patching my use case with `# nolint` for now.

**Comment by peterhurford** [comment]
@wlandau I've run into that exact issue several times, leading me just not to do left parentheses linting.

**Comment by shimash63174** [comment]
@peterhurford I run into the same issue, Have you added not_exception <- !(before_operator %in% c("!", ":", "["))

**Comment by martin-mfg** [comment]
The reason for lintr not liking `((` most probably is [section 2.2 of the tidyverse style guide](http://style.tidyverse.org/syntax.html#spacing), which states

> Place a space before `(`, except when it’s part of a function call.

The runaround pointed out by @wlandau is also caused by the style guide, because it is inconsistent on whether to put a space in between `((` or not.

Regarding these two problems in the tidyverse style guide, I already opened an issue on the style guide's GitHub page: tidyverse/style#66.

**Comment by kenahoo** [comment]
The tidyverse style guide has been updated with a fix:

https://github.com/tidyverse/style/commit/bde1f157da734f2def99e386edf2b9e3fb64a8e1

Can this ticket be reopened so `lintr` can match the style guide?  There is currently no solution for code like this:

```
list((1:2)^3, 4)  # "Place a space before ("
list( (1:2)^3, 4)  # "Do not place spaces around code in parentheses"
```
